sdk/servicebus/service-bus/src/receivers/sessionReceiver.ts

```diff
@@ -113,56 +122,68 @@ export class SessionReceiverImpl<ReceivedMessageT extends ReceivedMessage | Rece
    * @throws Error if an open receiver is already existing for given sessionId.
    */
   private constructor(
+    messageSession: MessageSession,
     context: ClientEntityContext,
     public receiveMode: "peekLock" | "receiveAndDelete",
     private _sessionOptions: CreateSessionReceiverOptions,
     retryOptions: RetryOptions = {}
   ) {
     throwErrorIfConnectionClosed(context.namespace);
     this._context = context;
+    this._sessionOptions;
     this.entityPath = this._context.entityPath;
     this._retryOptions = retryOptions;
+    this._messageSession = messageSession;
+    // By this point, we should have a valid sessionId on the messageSession
+    // If not, the receiver cannot be used, so throw error.
+    if (messageSession.sessionId == undefined) {
+      const error = new Error("Something went wrong. Cannot lock a session.");
```

Comment:
If we have a messageSession without a session id, then it means that we failed to get a hold of the session. Let's throw the `SessionCannotBeLocked` error here.

Reply:
Updated

Reply:
![image](https://user-images.githubusercontent.com/10452642/85638151-02d39300-b63a-11ea-9ac4-0ed0cb48c197.png)
The highlighted if-check is never hit, since `messageSession._init()` also checks for the same thing and throws a similar error much before. 

![image](https://user-images.githubusercontent.com/10452642/85638053-c1db7e80-b639-11ea-9711-82ffcb097a1d.png)